src/libraries/System.Runtime/tests/System.Reflection.Tests/DefaultBinderTests.cs

```diff
@@ -107,6 +107,17 @@ public static void DefaultBinderNamedParametersSkippedAndOutOfOrderTest()
             Assert.Equal("MethodMoreParameters", method.Name);
         }
 
+        [Fact]
+        public void InvokeWithIncorrectTargetTypeThrowsCorrectException()
+        {
+            Type t = typeof(Sample);
+            object incorrectInstance = Activator.CreateInstance(t);
+            MethodInvoker invoker = MethodInvoker.Create(typeof(Test).GetMethod(nameof(Test.TestMethod)));
+
+            TargetException ex = Assert.Throws<TargetException>(() => invoker.Invoke(obj: incorrectInstance, "NotAnInt"));
+            Assert.Equal("Object type Test does not match target type Sample.", ex.Message);
```

Comment:
although we've never localized these .NET Core libraries and have no plans to, in general we keep the option open -- that is why the strings are in resx files. asserting the English text would make that harder. What we usually do is to assert the string contains the things we want in it, in this case Test and Sample. Those will appear in any locale.

Reply:
@danmoseley - thanks for the pointer regarding the string - I've swapped out that part.  I wasn't sure about using the verbatim English text so that helps clear that up.

@buyaa-n - I'm assuming you mean that this file also needs modifying. 
 ttps://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Reflection/Core/Execution/MethodBaseInvoker.cs
I'll get to that as well later this week.  It was not immediately apparent what was going on here because the test is shared between AOT and Managed code tests but the underlying implementation appears to be different.